beacon_node/lighthouse_network/src/rpc/codec.rs

```diff
@@ -572,44 +572,37 @@ fn handle_rpc_request<E: EthSpec>(
         SupportedProtocol::BlocksByRangeV1 => Ok(Some(RequestType::BlocksByRange(
             OldBlocksByRangeRequest::V1(OldBlocksByRangeRequestV1::from_ssz_bytes(decoded_buffer)?),
         ))),
-        SupportedProtocol::BlocksByRootV2 => Ok(Some(RequestType::BlocksByRoot(
-            BlocksByRootRequest::V2(BlocksByRootRequestV2 {
-                block_roots: RuntimeVariableList::from_ssz_bytes(
-                    decoded_buffer,
-                    spec.max_request_blocks as usize,
-                )?,
-            }),
-        ))),
-        SupportedProtocol::BlocksByRootV1 => Ok(Some(RequestType::BlocksByRoot(
-            BlocksByRootRequest::V1(BlocksByRootRequestV1 {
-                block_roots: RuntimeVariableList::from_ssz_bytes(
-                    decoded_buffer,
-                    spec.max_request_blocks as usize,
-                )?,
-            }),
-        ))),
-        SupportedProtocol::BlobsByRangeV1 => {
-            let req = BlobsByRangeRequest::from_ssz_bytes(decoded_buffer)?;
-            let max_requested_blobs = req
-                .count
-                .saturating_mul(spec.max_blobs_per_block_by_fork(current_fork));
-            // TODO(pawan): change this to max_blobs_per_rpc_request in the alpha10 PR
-            if max_requested_blobs > spec.max_request_blob_sidecars {
-                return Err(RPCError::ErrorResponse(
-                    RpcErrorResponse::InvalidRequest,
-                    format!(
-                        "requested exceeded limit. allowed: {}, requested: {}",
-                        spec.max_request_blob_sidecars, max_requested_blobs
-                    ),
-                ));
-            }
-            Ok(Some(RequestType::BlobsByRange(req)))
+        SupportedProtocol::BlocksByRootV2 => {
+            let max_request_blocks = spec.max_request_blocks(current_fork);
+            Ok(Some(RequestType::BlocksByRoot(BlocksByRootRequest::V2(
+                BlocksByRootRequestV2 {
+                    block_roots: RuntimeVariableList::from_ssz_bytes(
+                        decoded_buffer,
+                        max_request_blocks,
+                    )?,
+                },
+            ))))
         }
+        SupportedProtocol::BlocksByRootV1 => {
+            let max_request_blocks = spec.max_request_blocks(current_fork);
+            Ok(Some(RequestType::BlocksByRoot(BlocksByRootRequest::V1(
+                BlocksByRootRequestV1 {
+                    block_roots: RuntimeVariableList::from_ssz_bytes(
+                        decoded_buffer,
+                        max_request_blocks,
+                    )?,
+                },
+            ))))
+        }
```

Comment:
nit:
```suggestion
        SupportedProtocol::BlocksByRootV1 => Ok(Some(RequestType::BlocksByRoot(
            BlocksByRootRequest::V1(BlocksByRootRequestV1 {
                block_roots: RuntimeVariableList::from_ssz_bytes(
                    decoded_buffer,
                    spec.max_request_blocks(current_fork),
                )?,
            }),
        ))),
```
we can just inline the `spec.max_request_blocks(current_fork)` wdyt Pawan?